File: Server/src/main/content/global/skill/production/cooking/recipe/pie/impl/WildPie.java
package content.global.skill.production.cooking.recipe.pie.impl;

import content.global.skill.production.cooking.recipe.pie.PieRecipe;
import core.game.node.item.Item;

/**
 * The Wild pie.
 */
public class WildPie extends PieRecipe {

    private static final Item UNCOOKED_PIE = new Item(7206);

    private static final Item BEAR_MEAT = new Item(2136);

    private static final Item CHOMPY_MEAT = new Item(2876);

    private static final Item RABBIT_MEAT = new Item(3226);

    private static final Item PART_ONE = new Item(7202);

    private static final Item PART_TWO = new Item(7204);

    @Override
    public Item getProduct() {
        return UNCOOKED_PIE;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{BEAR_MEAT, CHOMPY_MEAT, RABBIT_MEAT};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{PIE_SHELL, PART_ONE, PART_TWO, UNCOOKED_PIE};
    }

    @Override
    public boolean isSingular() {
        return false;
    }

}

File: Server/src/main/content/global/skill/production/cooking/recipe/pie/impl/WildPie.kt
package content.global.skill.production.cooking.recipe.pie.impl

import content.global.skill.production.cooking.recipe.pie.PieRecipe
import core.api.consts.Items
import core.game.node.item.Item

class WildPie : PieRecipe() {

    override fun getProduct(): Item {
        return UNCOOKED_PIE
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(BEAR_MEAT, CHOMPY_MEAT, RABBIT_MEAT)
    }

    override fun getParts(): Array<Item> {
        return arrayOf(PIE_SHELL, PART_ONE, PART_TWO, UNCOOKED_PIE)
    }

    override fun isSingular(): Boolean {
        return false
    }

    companion object {
        private val UNCOOKED_PIE = Item(Items.RAW_WILD_PIE_7206)
        private val BEAR_MEAT = Item(Items.RAW_BEAR_MEAT_2136)
        private val CHOMPY_MEAT = Item(Items.RAW_CHOMPY_2876)
        private val RABBIT_MEAT = Item(Items.RAW_RABBIT_3226)
        private val PART_ONE = Item(Items.PART_WILD_PIE_7202)
        private val PART_TWO = Item(Items.PART_WILD_PIE_7204)
    }
}
